lib/gates_cirq.h

```diff
@@ -0,0 +1,1114 @@
+// Copyright 2019 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GATES_CIRQ_H_
+#define GATES_CIRQ_H_
+
+#include <array>
+#include <cmath>
+#include <complex>
+#include <vector>
+
+#include "gate.h"
+
+namespace qsim {
+
+namespace Cirq {
+
+enum GateKind {
+  kI = 0,     // One-qubit identity gate.
+  kI2,        // Two-qubit identity gate.
+  kXPowGate,
+  kYPowGate,
+  kZPowGate,
+  kHPowGate,
+  kCZPowGate,
+  kCXPowGate,
+  krx,
+  kry,
+  krz,
```

Comment:
These may not be needed - see comments below.

Reply:
I didn't remove the rx/ry/rz gates. So those are still needed.